pkg/prove/proof.go

```diff
@@ -0,0 +1,187 @@
+package prove
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+
+	"github.com/celestiaorg/rsmt2d"
+	tmbytes "github.com/tendermint/tendermint/libs/bytes"
+	"github.com/tendermint/tendermint/pkg/consts"
+	"github.com/tendermint/tendermint/pkg/wrapper"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+	"github.com/tendermint/tendermint/types"
+)
+
+// TxInclusion uses the provided block data to progressively generate rows
+// of a data square, and then using those shares to creates nmt inclusion proofs
+// It is possible that a transaction spans more than one row. In that case, we
+// have to return two proofs.
```

Comment:
Orthogonal to this PR but doesn't this also support cases with more than 2 proofs?

Reply:
yeah it should. We don't have an explicit test for this, though. I ran the existing test which uses random data a few thousand times, and a multiple proofs never showed up, so I'm not entirely convinced it is actually working. Although that could partially be the tests fault. We still don't support returning proofs for txs that span more than two shares #621 so that makes it more difficult to test this.